src/dev/eslint/run_eslint_custom.ts

```diff
@@ -0,0 +1,303 @@
+      }, 0);
+
+      if (totalFindings > 0) {
+        process.exit(1);
+      }
+    } else {
+      const hadFailures = results.some((r) => (r.exitCode ?? 0) !== 0);
+
+      if (hadFailures) {
+        process.exit(1);
+      }
+    }
+  },
+  {
+    description: 'Run a custom eslint config on kibana packages',
+    flags: {
+      string: ['concurrency', 'formatter', 'eslint-config', 'filter'],
+      default: {
+        concurrency: 1,
+        'eslint-config': 'eslintrc.js',
+      },
+      help: `
+        --concurrency <n>       Number of parallel ESLint workers (default: CPU count)
+        --formatter <path>      Optional module path default-exporting a function to format results
+        --eslint-config <path>  Optional path to an eslint config (default: eslintrc.js)
+        --filter <path>         Optional filter: match package folder substring or exact package name
+      `,
+    },
+  }
+);
+
+function defaultFormatter(log: ToolingLog, results: EslintRunResult[]) {
+  // Build a simple summary and print it as-is
+  const summary = new Map<string, Map<string, number>>();
+
+  for (const r of results) {
+    if (!r || !Array.isArray(r.report)) continue;
+
+    let pkgMap = summary.get(r.name);
+
+    if (!pkgMap) {
+      pkgMap = new Map();
+      summary.set(r.name, pkgMap);
+    }
+
+    for (const file of r.report) {
+      const msgs = Array.isArray(file?.messages) ? file.messages : [];
+
+      for (const m of msgs) {
+        const key = m && typeof m.message === 'string' ? m.message : String(m?.ruleId || 'unknown');
+        pkgMap.set(key, (pkgMap.get(key) || 0) + 1);
+      }
+    }
+  }
+
+  for (const [name, counts] of summary) {
+    if (!counts || counts.size === 0) continue;
+
+    log.info(name);
+
+    for (const [msg, count] of counts) {
+      log.info(`${msg} ${count}`);
+    }
+
+    log.write('\n');
+  }
+}
+
+async function loadFormatterModule(modPath: string, log: ToolingLog): Promise<FormatterFn | null> {
+  const absoluteModulePath = path.isAbsolute(modPath) ? modPath : path.resolve(REPO_ROOT, modPath);
+
+  const isFormatterFn = (x: unknown): x is FormatterFn => typeof x === 'function';
+
+  const isDict = (x: unknown): x is Record<string, unknown> => typeof x === 'object' && x !== null;
+
+  try {
+    const module = await import(absoluteModulePath);
+    if (isDict(module)) {
+      const d = (module as { default?: unknown }).default;
+
+      if (isFormatterFn(d)) {
+        return d;
+      }
+    }
+  } catch (e) {
+    log.error(`Error in loading formatter ${e}`);
+  }
+
+  log.warning(
+    `Formatter module does not have a default exported function. Please export a default function from ${modPath}.`
+  );
+  return null;
+}
+
+async function runEslintOnEntry(
+  { name, directory }: Package,
+  opts: { json: boolean; stream: boolean; eslintConf: string }
+): Promise<EslintRunResult> {
+  const glob = `${directory}/**/*.{ts,tsx,js,jsx}`;
+
+  const args = [
+    'scripts/eslint',
```

Comment:
You can try to resolve and run the `eslint` binary directly instead of going through the kibana eslint script. I think that was set up as a convenience script, so we can shovel in files to it as arguments, for you this might be just an added layer of indirection, including babel register and security hardening.